clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/security/KubernetesNamedAccountCredentials.java

```diff
@@ -419,8 +429,20 @@ KubernetesNamedAccountCredentials build() {
         providerVersion = v1;
       }
 
-      if (StringUtils.isEmpty(kubeconfigFile)) {
-        kubeconfigFile = System.getProperty("user.home") + "/.kube/config";
+      if (StringUtils.isEmpty(kubeconfigFile)){
+        if (StringUtils.isEmpty(kubeconfigContents)) {
+          kubeconfigFile = System.getProperty("user.home") + "/.kube/config";
+        } else {
+          try {
+            File temp = File.createTempFile("kube", "config");
+            BufferedWriter writer = new BufferedWriter(new FileWriter(temp));
+            writer.write(kubeconfigContents);
+            writer.close();
```

Comment:
Could use this next time to make it read a bit cleaner: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html